### ISSUE TITLE
Default backend overwrites backends in rules

### ISSUE THREAD

**Issue by cten** [heptio/contour#405]
I am trying to deploy and ingress with a default backend that will catch any requests that don't match one of the "rules". 
I'm using contour version of master downloaded 2 days ago (docker image ID: 5d06382ae87e)  and Kubernetes 1.10.2

My ingress resource:
```
spec:
  backend:
    serviceName: kuard
    servicePort: 80
  rules:
  - host: test-gui
    http:
      paths:
      - backend:
          serviceName: test-gui
          servicePort: 80
        path: /
status:
  loadBalancer: {}
```

Envoy /routes output (I took out the white space and newlines to make it shorter)
```
"route_table_dump": {
 "name": "ingress_http",
 "virtual_hosts": [
  { "name": "*",
   "domains": [
    "*" ],
   "routes": [
    { "match": {
      "prefix": "/" },
     "route": {
      "cluster": "default/kuard/80" 
}}]},
  {"name": "test-gui",
   "domains": [
    "test-gui",
    "test-gui:80" ],
   "routes": [
    {"match": {
      "prefix": "/"},
     "route": {
      "cluster": "default/kuard/80" }
    }]}]}}]
```
As you can see kuard is being set as the backend for both routes.

**Comment by davecheney** [comment]
Confirmed. Thank you for reporting the issue. I'm working on a fix now.

**Comment by davecheney** [comment]
Closed by #446
